target_actionnetwork/sinks.py

```diff
@@ -93,6 +93,9 @@ def preprocess_record(self, record: dict, context: dict) -> dict:
     def upsert_record(self, record: dict, context: dict):
         state_updates = dict()
         if record:
+            if record.get("error"):
+                raise Exception(record.get("error"))
+
```

Comment:
this is okay but we still need to add a try except in `preprocess_record` and return it as {"error": str(e)} to be able to catch those in upsert_record